objects/src/accounts/code.rs

```diff
@@ -53,30 +73,44 @@ impl AccountCode {
 
     /// Returns a commitment to an account's public interface.
     pub fn root(&self) -> Digest {
-        self.method_tree.root().into()
+        self.procedure_tree.root().into()
     }
 
-    /// Returns the number of public interface methods defined for this account.
-    pub fn num_methods(&self) -> usize {
-        todo!()
+    /// Returns a reference to the [ModuleAst] backing the [AccountCode].
+    pub fn module(&self) -> &ModuleAst {
+        &self.module
     }
 
-    /// Returns true if a method with the specified root is defined for this account.
-    pub fn has_method(&self, _root: Digest) -> bool {
-        todo!()
+    /// Returns a reference to the procedure tree.
+    pub fn procedure_tree(&self) -> &SimpleSmt {
+        &self.procedure_tree
     }
 
-    /// Returns an account interface method at the specified index.
+    /// Returns the number of public interface procedures defined for this account.
+    pub fn num_procedures(&self) -> usize {
+        self.procedures.len()
+    }
+
+    /// Returns true if a procedure with the specified root is defined for this account.
+    pub fn has_procedure(&self, root: Digest) -> bool {
+        self.procedures.binary_search_by(|r| r.cmp(&root)).is_ok()
+    }
+
+    /// Returns a procedure digest for the procedure with the specified index.
     ///
     /// # Panics
     /// Panics if the provided index is out of bounds.
-    pub fn get_method_by_index(&self, _index: usize) -> &Program {
-        todo!()
+    pub fn get_procedure_by_index(&self, index: usize) -> Digest {
+        // index must be wihtin range
+        assert!(index < self.procedures.len());
+
+        // Return digest for the procedure
+        *self.procedures.get(index).unwrap()
     }
 
-    /// Returns an account interface method with the specified root or None if such method is not
-    /// defined for this account.
-    pub fn get_method_by_root(&self, _root: Digest) -> Option<&Program> {
-        todo!()
+    /// Returns the procedure index for the procedure with the specified root or None if such
+    /// procedure is not defined for this account.
+    pub fn get_procedure_index_by_root(&self, root: Digest) -> Option<usize> {
+        self.procedures.binary_search_by(|x| x.cmp(&root)).ok()
```

Comment:
Would binary sort work here correctly as we sort slightly differently in the constructor?

Reply:
updated to be consistent with constructor ordering.